### PR TITLE
Feature/redux

### PR DESCRIPTION
**Pull request by vigneshBS22**

### CODE REVIEW

File: Navigators/AuthNavigator.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export const HomeTabScreen = () => {
+  return (
+    <HomeTab.Navigator>
+      <HomeTab.Screen
+        name="Home"
+        component={HomeScreen}
+        options={{
+          headerShown: false,
+          tabBarLabel: 'Home',
```
Comment (joydeep33) [inline Navigators/AuthNavigator.js]: Tab label and name should be from a constants file

File: Navigators/AuthNavigator.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          tabBarIcon: ({color, size}) => (
+            <AntDesign name="home" color={color} size={size} />
+          ),
+        }}
+      />
+      <HomeTab.Screen
+        name="Anime"
```
Comment (joydeep33) [inline Navigators/AuthNavigator.js]: same as above

File: Navigators/AuthNavigator.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        component={AnimeScreen}
+        options={{
+          tabBarIcon: ({color, size}) => (
+            <AntDesign name="eyeo" color={color} size={size} />
+          ),
+        }}
+      />
+      <HomeTab.Screen
+        name="Games"
```
Comment (joydeep33) [inline Navigators/AuthNavigator.js]: same as above

File: Navigators/AuthNavigator.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+              size={size}
+            />
+          ),
+        }}
+      />
+      <HomeTab.Screen
+        name="Settings"
+        component={SettingScreen}
+        options={{
+          tabBarLabel: 'Settings',
```
Comment (joydeep33) [inline Navigators/AuthNavigator.js]: same as above

File: Navigators/LoginNavigators.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import {LoginTab, RegisterStack} from '../constants';
+
+const LoginTabScreen = () => {
+  return (
+    <LoginTab.Navigator>
+      <LoginTab.Screen
+        name="Login"
+        options={{
+          headerShown: false,
+          tabBarLabel: 'Login',
```
Comment (joydeep33) [inline Navigators/LoginNavigators.js]: same as above

File: Navigators/LoginNavigators.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+          ),
+        }}>
+        {props => <LoginScreen {...props} />}
+      </LoginTab.Screen>
+      <LoginTab.Screen
+        name="Register"
+        component={RegisterScreen}
+        options={{
+          headerShown: false,
+          tabBarLabel: 'Register',
```
Comment (joydeep33) [inline Navigators/LoginNavigators.js]: same as above

File: Navigators/LoginNavigators.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      />
+    </LoginTab.Navigator>
+  );
+};
+
+export const LoginStackScreen = () => {
+  return (
+    <RegisterStack.Navigator>
+      <RegisterStack.Screen
+        name="LoginScreen"
```
Comment (joydeep33) [inline Navigators/LoginNavigators.js]: same as above

File: Navigators/LoginNavigators.js [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+        options={{
+          headerShown: false,
+        }}>
+        {props => <LoginTabScreen {...props} />}
+      </RegisterStack.Screen>
+      <RegisterStack.Screen
+        name="Profile"
```
Comment (joydeep33) [inline Navigators/LoginNavigators.js]: same as above

File: Navigators/RootNavigator.js [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  const dispatch = useDispatch();
+  const {name} = useSelector(selectAuth);
+  const {
+    state: {color},
+  } = useColor();
+
+  // Handle user state changes
+  async function onAuthStateChanged(user) {
+    if (user) {
+      const docRef = firestore().collection('Users').doc(user.uid);
```
Comment (joydeep33) [inline Navigators/RootNavigator.js]: move this async operation to thunk

Comment (joydeep33) [inline Navigators/RootNavigator.js]: move this async operation to thunk

File: Navigators/RootNavigator.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    return subscriber; // unsubscribe on unmount
+  }, []);
+
+  return (
+    <NavigationContainer theme={color === 'light' ? MyTheme : MyDarkTheme}>
```
Comment (joydeep33) [inline Navigators/RootNavigator.js]: The color mode should be from the constants file.
e.g - const COLOR_MODE = {
LIGHT: 'light',
DARK: 'dark'
}

File: Navigators/RootNavigator.js [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+      {name === '' ? (
+        <AuthStack.Navigator>
+          <AuthStack.Screen
+            name="WelcomeScreen"
```
Comment (joydeep33) [inline Navigators/RootNavigator.js]: Screen names should be from constants file

File: Navigators/RootNavigator.js [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+            options={{headerShown: false}}
+            component={WelcomeScreen}
+          />
+          <AuthStack.Screen name="Login" options={{headerShown: false}}>
```
Comment (joydeep33) [inline Navigators/RootNavigator.js]: Same as above

File: Navigators/RootNavigator.js [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+            {props => <LoginStackScreen {...props} />}
+          </AuthStack.Screen>
+        </AuthStack.Navigator>
+      ) : (
+        <HomeStack.Navigator>
+          <HomeStack.Screen
+            name="HomeScreen"
```
Comment (joydeep33) [inline Navigators/RootNavigator.js]: same as above

File: components/Card.js [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+          right="7"
+          px="3"
+          py="1.5"
+          borderRadius={5}>
+          5*
+        </Center>
+      </Box>
+      <Center py={2}>
+        <Text fontWeight={'extrabold'} fontStyle={'italic'}>
+          Link Click
```
Comment (joydeep33) [inline components/Card.js:54]: Move constants in a separate file

File: components/GoogleIcon.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"></Path>
+          <Path
+            fill="#4CAF50"
+            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"></Path>
+          <Path
+            fill="#1976D2"
+            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"></Path>
+        </Icon>
+        <Text mt={2.5} bold fontSize={18} color={theme.text}>
+          Google
```
Comment (joydeep33) [inline components/GoogleIcon.js:25]: Move to constants file

File: ios/Mini_project/Info.plist [hunk 16]
```diff
@@ -1,55 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (joydeep33) [inline ios/Mini_project/Info.plist]: Why did you remove this Info.plist file?

File: screens/ProfileScreen/ProfileScreen.js [hunk 17]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import {Text, View, TouchableOpacity} from 'react-native';
+
+const ProfileScreen = ({navigation}) => {
+  return (
+    <View style={{flex: 1, alignItems: 'center', justifyContent: 'center'}}>
```
Comment (joydeep33) [inline screens/ProfileScreen/ProfileScreen.js]: Don't add inline styles

File: screens/RegisterScreen/RegisterScreen.js [hunk 18]
```diff
@@ -0,0 +1,122 @@
...
+
+const RegisterScreen = ({navigation}) => {
+  const {
+    state: {theme},
+  } = useColor();
+  return (
+    <NativeBaseProvider>
+      <Box mx={10} my={5} mt={20}>
+        <Text fontSize={40} bold color={theme.text}>
+          Sign Up
```
Comment (joydeep33) [inline screens/RegisterScreen/RegisterScreen.js:116]: move this to constants file

File: screens/RegisterScreen/RegisterScreen.js [hunk 19]
```diff
@@ -0,0 +1,122 @@
...
+          shadow={8}
+          mt="2"
+          w="90%"
+          p={3}
+          colorScheme="indigo"
+          mx={'5%'}
+          onPress={() => {
+            onSignUpButtonPress(details);
+          }}>
+          Signup
```
Comment (joydeep33) [inline screens/RegisterScreen/RegisterScreen.js:101]: move this to constants file

File: screens/WelcomeScreen/WelcomeScreen.js [hunk 20]
```diff
@@ -0,0 +1,24 @@
...
+    state: {theme},
+  } = useColor();
+  return (
+    <Center flex={1} bg={theme.bg}>
+      <Center px="3" py="1.5">
+        <Button
+          onPress={() => {
+            navigation.navigate('Login');
+          }}>
+          <Text fontSize={20}>Get Started</Text>
```
Comment (joydeep33) [inline screens/WelcomeScreen/WelcomeScreen.js:17]: Move this to constants file

File: utils/emailLogin.js [hunk 21]
```diff
@@ -0,0 +1,19 @@
+import auth from '@react-native-firebase/auth';
+
+export const OnLoginButtonPress = details => {
```
Comment (joydeep33) [inline utils/emailLogin.js]: This async operation should be performed using thunk

File: utils/emailSignup.js [hunk 22]
```diff
@@ -0,0 +1,33 @@
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+
+export const onSignUpButtonPress = details => {
```
Comment (joydeep33) [inline utils/emailSignup.js:4]: This async operation should be performed using thunk

File: utils/facebookLogin.js [hunk 23]
```diff
@@ -0,0 +1,43 @@
+import {LoginManager, AccessToken} from 'react-native-fbsdk-next';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+
+export async function onFacebookButtonPress() {
```
Comment (joydeep33) [inline utils/facebookLogin.js]: This async operation should be performed using thunk

File: utils/googleLogin.js [hunk 24]
```diff
@@ -0,0 +1,33 @@
+import {GoogleSignin} from '@react-native-google-signin/google-signin';
+import auth from '@react-native-firebase/auth';
+import firestore from '@react-native-firebase/firestore';
+import Config from 'react-native-config';
+
+GoogleSignin.configure({
+  webClientId: Config.GOOGLE_WEBCLIENT_ID,
+});
+
+export async function onGoogleButtonPress() {
```
Comment (joydeep33) [inline utils/googleLogin.js]: This async operation should be performed using thunk

**Comment by joydeep33** [comment]
**Feedback -** 

- Move constants to a separate constants file.
- Perform async logic using thunk by dispatching an action.
- Don't use inline styles unless it's necessary.
@vigneshBS22

File: ios/Mini_project/Info.plist [hunk 25]
```diff
@@ -1,117 +0,0 @@
...
-				<string>fb961116228116960</string>
-			</array>
-		</dict>
-	</array>
-	<key>CFBundleVersion</key>
-	<string>1</string>
-	<key>FacebookAppID</key>
-	<string>961116228116960</string>
-	<key>FacebookClientToken</key>
-	<string>412279dc30106b1859efedf9c53f4da3</string>
```
Comment (vigneshBS22) [inline ios/Mini_project/Info.plist]: @joydeep33 here
